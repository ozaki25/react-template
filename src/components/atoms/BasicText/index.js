import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function BasicText({ children }) {
  return <P>{children}</P>;
}

export const P = styled.p`
  color: #333;
  font-size: 16px;
`;

BasicText.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BasicText;
