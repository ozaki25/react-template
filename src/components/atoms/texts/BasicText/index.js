import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const BasicText = ({ children }) => <P>{children}</P>;

const P = styled.p`
  color: #333;
  font-size: 16px;
`;

BasicText.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BasicText;
