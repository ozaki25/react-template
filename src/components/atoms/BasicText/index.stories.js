import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';

import BasicText from '.';

const stories = storiesOf('atoms', module);

stories.add(
  'BasicText',
  withInfo('標準文字列')(() => <BasicText>{text('text', '標準文字列')}</BasicText>),
);
