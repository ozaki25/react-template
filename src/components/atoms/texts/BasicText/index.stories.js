import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';

import BasicText from '.';

const stories = storiesOf('atoms/texts', module);

stories.add(
  'BasicText/nomal',
  withInfo('標準文字列')(() => <BasicText>{text('text', '標準文字列')}</BasicText>),
);

stories.add(
  'BasicText/emphasis',
  withInfo('強調する文字列')(() => (
    <BasicText emphasis>{text('text', '強調する文字列')}</BasicText>
  )),
);

stories.add(
  'BasicText/red',
  withInfo('強調する赤い文字列')(() => (
    <BasicText red>{text('text', '強調する赤い文字列')}</BasicText>
  )),
);

stories.add(
  'BasicText/red-emphasis',
  withInfo('強調する赤い文字列')(() => (
    <BasicText emphasis red>
      {text('text', '強調する赤い文字列')}
    </BasicText>
  )),
);
