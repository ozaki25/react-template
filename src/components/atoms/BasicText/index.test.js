import React from 'react';
import { snapshot } from 'test/helpers';
import BasicText, { P } from '.';

snapshot('BasicText', <BasicText>テキスト</BasicText>);
snapshot('BasicText/P', <P>テキスト</P>);
