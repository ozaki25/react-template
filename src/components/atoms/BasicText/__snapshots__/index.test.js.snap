// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: BasicText 1`] = `
<styled.p>
  テキスト
</styled.p>
`;

exports[`should match snapshot: BasicText/P 1`] = `
.c0 {
  color: #333;
  font-size: 16px;
}

<p
  className="c0"
>
  テキスト
</p>
`;
